packages/contracts/contracts/YieldGate.sol

```diff
@@ -68,13 +70,27 @@ contract YieldGate {
 }
 
 contract BeneficiaryPool {
-    event Staked(address indexed beneficiary, address indexed supporter, uint256 amount);
+    event Staked(
+        address indexed beneficiary,
+        address indexed supporter,
+        uint256 amount,
+        uint256 timeout
```

Comment:
Minor: I might find `timeout` not to be the perfect naming as this suggests that the staking ends when the time is over. But only the lock ends, right? But I also have no really good ideas beyond `lockTimeout` or `lockedUntil`.

Reply:
Agree, renamed this event var and also the contract storage mapping to `lockTimeouts`.

I'm just a fan of succinct names, but succinctness shouldn't come at the price of comprehensibility ;)